### DISCUSSION TITLE
How to Process .properties Files?

### DISCUSSION THREAD

**Discussion by anthumchris** [parcel-bundler/parcel#5855] [Q&A]
Could someone kindly point me in the right direction to enable or write a custom Transform module to process key/value `.properties` files?  I'm trying to move a front-end build from WebPack to Parcel, and the project has a bunch of `.properties` files that cause Parcel to throw an error because of a non-existent transformer available.

**`example.properties`**
```
prop.foo = Foo
prop.bar = Bar
```

**Comment by mischnic** [comment]
The transformer could look something like this (depending on what the generated JS should look like):
```js
// @flow

const {Transformer} = require('@parcel/plugin');

module.exports = new Transformer({
  async transform({asset}) {
    let input = await asset.getCode();
    let parsed = something(input); // a function to convert into JSON
    asset.type = 'js';
    asset.setCode(`export default ${JSON.stringify(parsed)};`);
    return [asset];
  },
});
```
